experimental/hip/hip_buffer.c

```diff
@@ -0,0 +1,167 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "experimental/hip/hip_buffer.h"
+
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "iree/base/api.h"
+#include "iree/base/tracing.h"
+
+typedef struct iree_hal_hip_buffer_t {
+  hipDeviceptr_t device_ptr;
```

Comment:
Can we put `device_ptr` to be near `host_ptr`? They are both pointers so a bit easier to read.